### PR TITLE
Fix DPT decoder bugs for Scenic parity

### PR DESCRIPTION
**Pull request by bingyic**
## Summary

Fixes multiple bugs in `pytorch/decoders.py` to achieve numerical parity (max diff < 1e-4) with the Scenic/Flax reference implementation.

### Changes

1. **`DPTHead`**: Add `output_activation` parameter (default `False`).
   When `True`, applies `F.relu()` after project conv, matching Scenic.

2. **`DepthDecoder`**: Replace with classification-based depth prediction:
   - `nn.Linear(channels, num_depth_bins)` head
   - `bin_centers` buffer via `torch.linspace(min_depth, max_depth, num_depth_bins)`
   - Forward: `relu(logits) + min_depth` → normalize → `einsum(probs, bin_centers)`

3. **`ReassembleBlocks`**: Use `F.gelu(x, approximate='tanh')` to match JAX default.

4. **ConvTranspose kernel**: Apply 180° spatial flip during Flax→PyTorch conversion.

5. **`load_decoder_weights()`**: Unified weight loading from Scenic `.zip` checkpoints
   with auto-detection and key remapping for all decoder types:
   `pixel_segmentation`, `pixel_depth_classif`, `pixel_normals` → `head`

### Verification

All three decoder types verified for numerical parity against Scenic reference:
- Normals: max diff < 1e-4 ✅
- Depth (classification, 256 bins): max diff < 1e-4 ✅
- Segmentation: max diff < 1e-4 ✅

### CODE REVIEW

File: pytorch/decoders.py [hunk 1]
```diff
@@ -18,13 +18,19 @@
 This module provides a shared DPT backbone (ReassembleBlocks + fusion) and
 task-specific decoder subclasses for segmentation, depth, and surface normals.
 
+The implementation mirrors the Scenic/Flax reference at:
+  scenic/projects/dense_features/models/decoders.py
```
Comment (koertchen) [inline pytorch/decoders.py]: please remove internal code references

File: pytorch/decoders.py [hunk 2]
```diff
@@ -87,7 +93,7 @@ def forward(
       residual = self.residual_unit(residual)
       x = x + residual
     x = self.main_unit(x)
-    # Upsample 2x with align_corners=True
+    # Upsample 2x with align_corners=True (matches Scenic reference)
```
Comment (koertchen) [inline pytorch/decoders.py]: same here

File: pytorch/decoders.py [hunk 3]
```diff
@@ -129,33 +146,49 @@ def forward(
...
 class DPTHead(nn.Module):
-  """Shared DPT head that produces dense feature maps from ViT intermediates.
+  """Shared DPT backbone: ReassembleBlocks + convs + fusion + project.
 
-  This module performs the common operations for all decoder types:
-  1. Reassembles intermediate ViT features at multiple scales.
-  2. Projects them to a common channel dimension.
-  3. Fuses them bottom-up through residual fusion blocks.
+  Matches the Scenic DPTHead from
+  ``research/vision/scene_understanding/imsight/modules/dpt.py``.
```
Comment (koertchen) [inline pytorch/decoders.py]: same here

File: pytorch/decoders.py [hunk 4]
```diff
@@ -226,126 +258,309 @@ def __init__(
...
       ).squeeze(1)
     return depth_map.unsqueeze(1)
 
 
 class NormalsDecoder(Decoder):
-  """Decoder for surface normals prediction."""
+  """Decoder for surface normal estimation.
+
+  Outputs normal map of shape ``(B, 3, H, W)``.
+  Note: outputs are NOT L2-normalized by default, matching the Scenic
```
Comment (koertchen) [inline pytorch/decoders.py]: same here
